code.js

```diff
@@ -1,3 +1,68 @@
 function dijkstra(graph, sourceNode) {
-    return [];
+    
+    let dist = [];
+    let distance = [];
+    let marked = [];
+    for (let i = 0; i < graph.length; i++) { //set all nodes esitmated dist to 1000 (which will be our infinity)
+        dist[i] = 1000;
+        marked[i] = false;
+    }
+    
+    dist[sourceNode] = 0; //set estimated dist to 0 for the sourceNode
+    let trueCount = 0;
+    //loopNumber = 1;
+    //console.log("befoere the first loop: " + dist);
+    while (trueCount < marked.length) {
+        //console.log("loop " + loopNumber);
+        let lowest = 1000; //set lowest to our infinity
+        let lowestIndex = 0;
+        for (let i = 0; i < dist.length; i++) { //find the node with the lowest estimated dist that is not marked
+            for (let j = 0; j < marked.length; j++) {
```

Comment:
You can make this more efficient by simply keeping track of the unmarked nodes instead of the marked ones.